### PR TITLE
Hossein rejected ownership banner #628

### LINKED ISSUE #628: If a request ownership is rejected a banner should appeared in the record indicating it
In function requestOwnerhsip() of Record.vue (https://github.com/FAIRsharing/fairsharing.github.io/blob/dev/src/views/Records/Record.vue#L303), when the value of a request of ownership has value "rejected" (claim === "rejected"), the banner needs to be properly updated. I think there are two options:
1) Have same banner note than when there is an error: "Sorry, your request to claim this record failed. Please contact us.", i.e. if (claim.error || claim === "rejected"){
                _module.error = "Sorry, your request to claim this record failed. Please contact us.";
                _module.canClaim = false;
...
2) Have a different third behaviour

Maybe, the option 1) is the easiest one to code. Then @allysonlister , do you agree with current message that appears in 1)?

### PR DESCRIPTION
**Pull request by hoseinmirian**
Now we have approved and rejected as well as under review which was already implemented. to test:

1- please ask for ownership of a record
2- after asking for ownership refresh, you should see the under review banner. 
3- once reject it and go back and refresh the record page to see the banner
4- choose another random record and ask for ownership and this time approves and get back to record and refresh.

### CODE REVIEW

File: src/views/Records/Record.vue [hunk 1]
```diff
@@ -571,17 +591,21 @@ export default {
         try {
           const claim = await client.canClaim(recordID, _module.user().credentials.token);
           if (claim.error) {
-            if (claim.error.response.data.existing) {
+            if (claim.error.response.data.existing && claim.error.response.data.status==='pending') {
```
Comment (knirirr) [inline src/views/Records/Record.vue]:
```suggestion
            if (claim.error.response.data.existing && claim.error.response.data.status === 'pending') {
```

**Comment by allysonlister** [comment]
Yes, I needed to refresh in order to see the banners, but it is working. Some questions and notes:

1. Please update the rejection banner message, as "Your request to maintain this record is rejected!" is a bit shouty ;-) How about "Your claiming request has been declined. Please get in touch with us if you have any questions." 
2. Please update the initial banner message from "You have already requested to maintain this record. We will be getting back to you between 48 and 72h." to "You have already submitted an ownership request for this record. We will get back to you within a few working days."
3. Please update the approval message from "Your request to maintain this record is approved." to "Your request to maintain this record has been approved."
4. Is there any way to have the banner appear without refreshing the page, after step (1) in your comment above? 
5. It seems as though, for a few seconds, a floating comment in a green box appears just after I perform step (1) but the netlify border obscures most of it. Is something being said there? And if it is relevant, can it be said higher up on the page?
greenfloat

Separate to this particular ticket, I still find the oddness surrounding the "#" very hard to navigate when testing, as most hyperlinks don't work. Is there any way to fix this (e.g. I cannot refresh a page, because it always says that the page does not exist when I do)?

Thanks!

**Review by allysonlister — CHANGES_REQUESTED** [review]
Please see my previous comment for summary of change requests. Thanks!

**Comment by knirirr** [comment]
> Separate to this particular ticket, I still find the oddness surrounding the "#" very hard to navigate when testing, as most hyperlinks don't work. Is there any way to fix this (e.g. I cannot refresh a page, because it always says that the page does not exist when I do)?

I've changed the Netlify builds from 'hash' to 'history' mode but I'm not sure that will fix it.

File: src/views/Records/Record.vue [hunk 2]
```diff
@@ -571,17 +591,21 @@ export default {
...
+            if (claim.error.response.data.existing && claim.error.response.data.status === 'pending') {
               let maintainer = _module.getField("maintainers").filter(maintainer => maintainer.username === _module.user().credentials.username);
               if (maintainer.length === 0) {
                 _module.alreadyClaimed = true;
               }
             }
+            else _module.ownershipApproved = !(claim.error.response.data.status === 'rejected');
             _module.canClaim = false;
-          } else {
+          }else {
```
Comment (knirirr) [inline src/views/Records/Record.vue]:
```suggestion
          } 
          else {
```

**Comment by hoseinmirian** [comment]
1-done
2- done
3- done
4-done
5- no idea what's going on. does not happen to me. I think its trivial and suggest keep an eye on it when its merged to dev so if this happens again please open a new ticket. tnx

**Comment by knirirr** [comment]
The green box discussed above is a "toast" which displays the message "Thank you for claiming this record. We will get back to you within 48-72 hours." There will be no banner at this point but the claim menu is disabled so the user can't claim again. IIRC when this was discussed long ago a toast was requested rather than a banner, as too many banners might prove a nuisance. But, if a user leaves the page and comes back they might forget why that claim menu is disabled, hence the banner appearing later.

**Review by knirirr — APPROVED** [review]

**Comment by allysonlister** [comment]
Thanks both of you!

@knirirr - thanks for the green box information - I just couldn't see all of it, as scrolling did not make more appear and the netlify banner was in the way. Will check the updated page now.

**Comment by allysonlister** [comment]
Sorry, but I can't see any changes made in the netlify build - I tried ctrl-shift-R for a full refresh, but of course that isn't possible with the netlify build as I can't reload pages anymore in them - I just get page not found errors.

**Comment by hoseinmirian** [comment]
> Sorry, but I can't see any changes made in the netlify build - I tried ctrl-shift-R for a full refresh, but of course that isn't possible with the netlify build as I can't reload pages anymore in them - I just get page not found errors.

please use hash next to the id of your record like: 
https://deploy-preview-1403--fairsharing.netlify.app/#3

and then check the changes

**Comment by knirirr** [comment]
A refresh will be possible if dev is now merged into this branch.

**Comment by hoseinmirian** [comment]
> A refresh will be possible if dev is now merged into this branch.

do I have to push anymore in the branch?

**Comment by knirirr** [comment]
> do I have to push anymore in the branch?

Yes; you need:

```
git checkout dev
git pull
git checkout Hossein-rejected-ownership-banner-#628
git merge dev
git commit -a
git push
```

**Comment by hoseinmirian** [comment]
@allysonlister now the links can be accessed without #. so please go ahead. tnx

**Review by allysonlister — APPROVED** [review]
Thank you - all seems to be working well and as expected - and so much easier now that the refresh etc stuff has been sorted  - thanks to @hoseinmirian and @knirirr !

**Pull request merged**